### PR TITLE
#54 Buyer should be able to get a single product

### PR DESCRIPTION
**Pull request by jkarenzi**
#### What does this PR do?
Adds an endpoint where a buyer can retrieve a specific product

#### Description of Task to be completed?

- [x] Implement an API endpoint for retrieving a single product
- [x] Implement appropriate error handling and responses
- [x] Write comprehensive unit tests and documentation

#### How should this be manually tested?
- After cloning the repo, get a single product by accessing the api/v1/buyer/get_product/:id route and passing in the product id in the route parameters

#### Any background context you want to provide?
- This task depends on the manage product feature

####Screenshots
swagger
test 1
test 2

#### What are the relevant pivotal trackers?
#54

### CODE REVIEW

File: src/__test__/buyerController.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import request from 'supertest';
```
Comment (TresorXavier) [inline src/__test__/buyerController.test.ts:1]: Look great 👍

**Review by 13XAVI — APPROVED** [review]

**Review by EddyShimwa — APPROVED** [review]
Looks Good! 👍

**Review by wayneleon1 — APPROVED** [review]
that's cool

**Review by bertrandshema — APPROVED** [review]
Looks good to be merged

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**